### PR TITLE
fix: [PL-40073]: Add delegate info like tags, identifier to register audit event

### PR DESCRIPTION
**Pull request by jennyjames21**
## Describe your changes
Summary
- Add delegate info in the delegate register event

Test
- Test delegate audit entry 
- Unit test

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50888)

### CODE REVIEW

**Comment by jennyjames21** [comment]
trigger uts

**Comment by jennyjames21** [comment]
trigger uts

**Review by AnupamIO — APPROVED** [review]

**Comment by jennyjames21** [comment]
trigger uts

**Comment by raghuAtWings** [comment]
@jennyjames21 can you post a screenshot of how audit events look in case there is new registration or if there is any update on the tags?

**Comment by jennyjames21** [comment]
image

**Comment by jennyjames21** [comment]
trigger uts

**Comment by jennyjames21** [comment]
> @jennyjames21 can you post a screenshot of how audit events look in case there is new registration or if there is any update on the tags?

image

**Comment by jennyjames21** [comment]
trigger uts

File: 400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java [hunk 1]
```diff
@@ -2618,17 +2611,24 @@ public DelegateRegisterResponse register(final DelegateParams delegateParams, fi
...
+            .hostName(delegateParams.getHostName())
+            .orgIdentifier(orgIdentifier)
+            .projectIdentifier(projectIdentifier)
+            .description(delegateParams.getDescription())
+            .delegateType(delegateParams.getDelegateType())
+            .tags(isNotEmpty(delegateParams.getTags()) ? new HashSet<>(delegateParams.getTags()) : null)
+            .tokenName(delegateTokenName.orElse(null))
+            .build();
+
+    String delegateGroupId = delegateParams.getDelegateGroupId();
```
Comment (gn-harness) [inline 400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java:2626]: any particular reason to move this from above?

**Review by AnupamIO — APPROVED** [review]

File: 420-delegate-service/src/test/java/io/harness/delegate/outbox/DelegateOutboxEventHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+                                  .eventData(objectMapper.writeValueAsString(delegateUnregisterEvent))
+                                  .resource(delegateUnregisterEvent.getResource())
+                                  .createdAt(Long.parseLong(randomNumeric(5)))
+                                  .build();
+    final ArgumentCaptor<AuditEntry> auditEntryArgumentCaptor = ArgumentCaptor.forClass(AuditEntry.class);
+    when(auditClientService.publishAudit(any(), any())).thenReturn(true);
+    delegateOutboxEventHandler.handleDelegateUnRegisterEvent(outboxEvent);
+    verify(auditClientService, times(1)).publishAudit(auditEntryArgumentCaptor.capture(), any());
+    AuditEntry auditEntry = auditEntryArgumentCaptor.getValue();
+    assertEquals(Action.DELETE, auditEntry.getAction());
```
Comment (gn-harness) [inline 420-delegate-service/src/test/java/io/harness/delegate/outbox/DelegateOutboxEventHandlerTest.java:133]: missed "assertEquals(auditEntry.getNewYaml(), NGYamlUtils.getYamlString(delegateSetupDetails));"

**Review by gn-harness — APPROVED** [review]

File: 400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java [hunk 3]
```diff
@@ -2596,13 +2596,6 @@ public DelegateRegisterResponse register(final DelegateParams delegateParams, fi
 
     log.info("Registering delegate for Hostname: {} IP: {}", delegateParams.getHostName(), delegateParams.getIp());
 
-    String delegateGroupId = delegateParams.getDelegateGroupId();
-    if (isBlank(delegateGroupId) && isNotBlank(delegateParams.getDelegateGroupName())) {
```
Comment (xingchi-jin) [inline 400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java:2600]: removing this, meaning CG will not create delegate group.
Is this intentional ?

**Review by xingchi-jin — APPROVED** [review]

File: 420-delegate-service/src/test/java/io/harness/delegate/outbox/DelegateOutboxEventHandlerTest.java (hunk 2 above)
Comment (jennyjames21) [reply 420-delegate-service/src/test/java/io/harness/delegate/outbox/DelegateOutboxEventHandlerTest.java:133]: delete dont have yaml

File: 400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java (hunk 1 above)
Comment (jennyjames21) [reply 400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java:2626]: delegate setup details are needed for registering delegate group audit entry

**Comment by jennyjames21** [comment]
trigger messagecheck

**Pull request merged**